src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
+
+  catch(exception: HttpException, host: ArgumentsHost) {
+    let err = exception['response'];
+    err = err.response ? err.response : JSON.stringify(err);
+    const context = host.switchToHttp();
+    const response = context.getResponse<Response>();
+    const status = exception.getStatus();
+    this.logger.error(
+      `Message : ${err}`,
+      `${context.getRequest().method} ${context.getRequest().url}`,
+    );
+    const error =
+      typeof exception.getResponse() === 'string'
+        ? exception.getResponse()
+        : exception.getResponse() &&
+            typeof exception.getResponse()['message'] === 'object'
+          ? exception.getResponse()['message'].join(', ')
+          : exception.getResponse()['message'];
```

Comment:
PullProofAI

The conditional check for different types of responses can be simplified and made more readable using a ternary operator.

```diff
- typeof exception.getResponse() === 'string'
-   ? exception.getResponse()
-   : exception.getResponse() &&
-       typeof exception.getResponse()['message'] === 'object'
-     ? exception.getResponse()['message'].join(', ')
-     : exception.getResponse()['message'];
+ const responseMessage = 
+   typeof exception.getResponse() === 'string' ?
+     exception.getResponse() :
+     Array.isArray(exception.getResponse()?.message) ?
+       exception.getResponse().message.join(', ') :
+       exception.getResponse().message;
```

### Review Summary

The code changes in the `HttpExceptionFilter` class look good overall, but there are some improvements suggested to enhance readability and prevent potential runtime errors.


<!-- This is an auto-generated comment by PullProofAI -->